tools/cldr-code/src/main/java/org/unicode/cldr/util/DeferredTranscript.java

```diff
@@ -0,0 +1,72 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * lazily-calculates the transcript. Thread safe to call get() multiple times once commit() is
+ * called and before clear() is called.
+ */
+public class DeferredTranscript implements Supplier<String> {
+
+    private final class FormatEntry {
+        public final CharSequence fmt;
+        public final Object[] args;
+
+        public FormatEntry(String fmt, Object[] args) {
```

Comment:
Object... is better, because then it takes both a list and an array.

Reply:
Or make this private

Reply:
It actually is a private object. this is the internal storage for the deferred format.

The public API is `DeferredTranscript.add()`` which does take `Object...`